### PR TITLE
DOC: update tight layout guide with set_in_layout

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

The tight layout guide was obsolete with respect to what artists are automatically included in the layout calc...

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
The CI issues are not this PR (see https://github.com/matplotlib/matplotlib/pull/18101).  The built docs for this PR are here:  https://42418-1385122-gh.circle-artifacts.com/0/doc/build/html/tutorials/intermediate/tight_layout_guide.html#caveats

**Comment by QuLogic** [comment]
I think it's indented too much (and thus in a `blockquote`); can you remove the extra first space?

File: tutorials/intermediate/tight_layout_guide.py [hunk 1]
```diff
@@ -152,13 +152,13 @@ def example_plot(ax, fontsize=12):
 # Caveats
 # =======
 #
-#  * :func:`~matplotlib.pyplot.tight_layout` only considers ticklabels, axis
-#    labels, and titles. Thus, other artists may be clipped and also may
-#    overlap.
+#  * :func:`~matplotlib.pyplot.tight_layout` by default considers all artists
+#    on the axes.  To remove an artist from the layout calculation you can call
```
Comment (QuLogic) [inline tutorials/intermediate/tight_layout_guide.py:156]:
```suggestion
#    on the Axes. To remove an artist from the layout calculation you can call
```

File: tutorials/intermediate/tight_layout_guide.py [hunk 2]
```diff
@@ -152,13 +152,13 @@ def example_plot(ax, fontsize=12):
...
+#    `~.Artist.set_in_layout`.
```
Comment (QuLogic) [inline tutorials/intermediate/tight_layout_guide.py:157]: I would keep the reference to `Artist`:
```suggestion
#    `.Artist.set_in_layout`.
```

**Comment by jklymak** [comment]
New artifact: https://42441-1385122-gh.circle-artifacts.com/0/doc/build/html/tutorials/intermediate/tight_layout_guide.html#caveats

**Pull request merged**